changed snapshot placeholder and updated copy

Israellund opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #6486 

## Description of Changes
-Added a more specific placeholder to Snapshot input and added more copy to direct the user on what not to input. This copy was written and cleared by Jared. 

## "How We Fixed It"
<!-- Brief description of solution, if bug, to be added once issue is resolved. -->
-changed placeholder and added more copy

mzparacha reviewed (approved):
LGTM

CowMuon commented on packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Snapshots/Snapshots.tsx:
```diff
@@ -60,14 +60,16 @@ const Snapshots = () => {
           You can connect multiple Snapshot spaces to keep your voting
-          conversations all in one place.
+          conversations all in one place. Note: only input the name of your
+          snapshot space, not the full url.
```
Are we able to handle either for them? Would prefer if this were handled intelligently, rather than putting it on the user.

Israellund replied:
@CowMuon Yes I can set up input validation so that they aren't able to input an address with http or https. Good catch.

mzparacha commented on packages/commonwealth/client/scripts/views/components/LinksArray/types.ts:
```diff
@@ -42,5 +42,5 @@ export type LinksArrayProps = {
 export type LinksArrayHookProps = {
   initialLinks: Link[];
-  linkValidation?: z.ZodEffects<z.ZodString, string, string> | z.ZodString;
+  linkValidation?: typeof snapshotValidationSchema;
```
This will break the webhooks input array. I suggest we undo the last commit. I believe its not needed as we already have the snapshot validation to check for `http/s` logic

mzparacha commented on packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Snapshots/Snapshots.tsx:
```diff
@@ -68,6 +68,7 @@ const Snapshots = () => {
+          placeholder="examplesnapshotspace.eth"
           links={snapshots}
           onLinkAdd={onLinkAdd}
           onLinkUpdatedAtIndex={onLinkUpdatedAtIndex}
```
Looks like the old copy update got missed.

mzparacha commented on packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Snapshots/validation.ts:
```diff
@@ -1,13 +1,10 @@
-);
+const snapshotValidationSchema = z
+  .string()
+  .refine((space) => /^[a-zA-Z0-9-.]+\.eth$|^[a-zA-Z0-9-]+\.xyz$/.test(space), {
```
Minor: there is a missing `.` here for the `.xyz` part of the regex, we need this to allow multiple `.` in the `.xyz` domain

```diff
- /^[a-zA-Z0-9-.]+\.eth$|^[a-zA-Z0-9-]+\.xyz$/
+ /^[a-zA-Z0-9-.]+\.eth$|^[a-zA-Z0-9-.]+\.xyz$/
```

Also as a good convention, we should escape literal hyphen chars, so the regex becomes something like this

```diff
+ /^[a-zA-Z0-9\-.]+\.eth$|^[a-zA-Z0-9\-.]+\.xyz$/
```

mzparacha reviewed (commented):
Looks good, some minor changes suggested above.

Israellund commented:
@mzparacha at the request of Forest @CowMuon, I changed the logic to handle truncating the Snapshot address for the user instead of advising them through input validation. Now if an address like `https://snapshot.org/#/commonspacetester.eth` is entered it is automatically split and is saved as `commonspacetester.eth`. I have also reverted back to the original input validation advising that the address end in .eth or .xyz. Because of this new logic change I have taken out the newer copy change that was in a previous commit. 

https://github.com/hicommonwealth/commonwealth/assets/69872984/e8cef227-007b-4caa-bd2c-e584e40b4cfe
